tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -20,6 +29,281 @@ def test_hyperopt_parameters(use_hyperband):
+    study = optuna.create_study()
+    dist = optuna.distributions.UniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=dist) for i in range(1, 8)]
+
+    trial = frozen_trial_factory(8)
+    sampler = TPESampler(n_startup_trials=5, seed=0)
+    with patch("optuna.Study.get_trials", return_value=past_trials[:4]):
+        with patch.object(
+            optuna.samplers.random.RandomSampler, "sample_independent", return_value=1.0
+        ) as sample_method:
+            sampler.sample_independent(study, trial, "param-a", dist)
+    assert sample_method.call_count == 1
+    sampler = TPESampler(n_startup_trials=5, seed=0)
+    with patch("optuna.Study.get_trials", return_value=past_trials[:5]):
+        with patch.object(
+            optuna.samplers.random.RandomSampler, "sample_independent", return_value=1.0
+        ) as sample_method:
+            sampler.sample_independent(study, trial, "param-a", dist)
+    assert sample_method.call_count == 0
+
+
+def test_sample_independent_misc_arguments() -> None:
+    study = optuna.create_study()
+    dist = optuna.distributions.UniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=dist) for i in range(1, 8)]
+
+    # Prepare a trial and a sample for later checks.
+    trial = frozen_trial_factory(8)
+    sampler = TPESampler(n_startup_trials=5, seed=0)
+    with patch("optuna.Study.get_trials", return_value=past_trials):
+        suggestion = sampler.sample_independent(study, trial, "param-a", dist)
+
+    # Test misc. parameters.
+    sampler = TPESampler(n_ei_candidates=13, n_startup_trials=5, seed=0)
+    with patch("optuna.Study.get_trials", return_value=past_trials):
+        assert sampler.sample_independent(study, trial, "param-a", dist) != suggestion
+
+    sampler = TPESampler(gamma=lambda _: 5, n_startup_trials=5, seed=0)
+    with patch("optuna.Study.get_trials", return_value=past_trials):
+        assert sampler.sample_independent(study, trial, "param-a", dist) != suggestion
+
+    sampler = TPESampler(
+        weights=lambda i: np.asarray([i * 0.11 for i in range(7)]), n_startup_trials=5, seed=0
+    )
+    with patch("optuna.Study.get_trials", return_value=past_trials):
+        assert sampler.sample_independent(study, trial, "param-a", dist) != suggestion
+
+
+def test_sample_independent_uniform_distributions() -> None:
+    study = optuna.create_study()
+
+    # Prepare sample from uniform distribution for cheking other distributions.
+    uni_dist = optuna.distributions.UniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=uni_dist) for i in range(1, 8)]
+    trial = frozen_trial_factory(8)
+    sampler = TPESampler(n_startup_trials=5, seed=0)
+    with patch("optuna.Study.get_trials", return_value=past_trials):
+        uniform_suggestion = sampler.sample_independent(study, trial, "param-a", uni_dist)
+    assert 1.0 <= uniform_suggestion <= 100.0
```

Comment:
A detail but `high` is expected to be exclusive https://github.com/optuna/optuna/blob/master/optuna/distributions.py#L110 so how about the follwing?
```suggestion
    assert 1.0 <= uniform_suggestion < 100.0
```